### ISSUE TITLE
Create a CONTRIBUTING.md

### ISSUE THREAD

**Issue by personalizedrefrigerator** [xournalpp/xournalpp#3065]
**Is your feature request related to a problem? Please describe.**
It would be really helpful to have a `CONTRIBUTING.md` file [like this one in `circuitikz`](https://github.com/circuitikz/circuitikz/blob/master/CONTRIBUTING.md) (thanks for this @Rmano!)

**Describe the solution you'd like**
I would like a `CONTRIBUITING.md` file to be created with information useful for new contributors, probably containing the following:
 * How to submit a pull request (and how to expect the review process to work).
 * General `git` instructions relevant to contributing to xournalpp (or a link to a good resource)!
 * Examples explaining how to add tests to the test suite.
 * A link to our coding conventions
 * An FAQ

**Describe alternatives you've considered**
 * Having this on the website
 * Having this in the Wiki

**Comment by createyourpersonalaccount** [comment]
This is a good idea, also a high level overview of the file hierarchy, or at least some of the files, might be good (or is that covered by the autogenerated documentation?)

**Comment by rolandlo** [comment]
> This is a good idea, also a high level overview of the file hierarchy, or at least some of the files, might be good (or is that covered by the autogenerated documentation?)

What file hierarchy do you mean? The file hierarchy of the source code or the file hierarchy of the install on various platforms? I have planned to detail the latter one on the [website](https://xournalpp.github.io).

**Comment by rolandlo** [comment]
> **Describe alternatives you've considered**
> 
>     * Having this on the website
> 
>     * Having this in the Wiki

The [website](https://xournalpp.github.io) is designed for the average user. Technical stuff like how to submit a PR and how to use git have no place there in my opinion. I am fine with the solution of creating `CONTRIBUTING.md`. It should be linked in the Readme or in the Wiki, though, so it can be found without browsing the files from the repository.

**Comment by createyourpersonalaccount** [comment]
> > This is a good idea, also a high level overview of the file hierarchy, or at least some of the files
>
> What file hierarchy do you mean? The file hierarchy of the source code or the file hierarchy of the install on various platforms? I have planned to detail the latter one on the [website](https://xournalpp.github.io).

How Xournal++ works is a mystery for me as a beginner. For instance, I may wish to develop an infinite-page feature (instead of A4). Roughly, which files should I be looking at? Of course that's a major feature and probably many files are affected, but that's the sort of thing I'm asking for, an overview of what source code files are responsible for.

**Comment by Technius** [comment]
That would be more appropriate for an ["ARCHITECTURE" file](https://matklad.github.io//2021/02/06/ARCHITECTURE.md.html), which is related to, but not the same as, CONTRIBUTING.

**Comment by idotobi** [comment]
I really like the idea of a `Contributing.md`. This could also be used to point to a `COPYRIGHT.md` and explain the context of it.
I've just bean reading up on copyright in OpenSource Projects and I think that the [recommendation of the Linux Foundation](https://www.linuxfoundation.org/blog/copyright-notices-in-open-source-software-projects/) seems very reasonable.

TL;DR: `Copyright The XYZ Authors` unless required differently

A more in depth discussion can be found in [this blogpost](https://opensource.com/article/20/10/copyright-notices-open-source-software). This could then also be used for debian packaging.

It might also be reasonable to include a "Developer Certificate of Origin" (https://developercertificate.org/) or something similar to the `Contributing.md`, but I'm not sure if this is really required.

**Comment by idotobi** [comment]
I created an issue for the `COPYRIGHT.md` file at https://github.com/xournalpp/xournalpp/issues/3103.

I think the "Developer Certificate of Origin" still makes sense to be discussed for the `Contributing.md` file discussion.
